core/app/services/spree/export/google_rss.rb

```diff
@@ -0,0 +1,90 @@
+require 'nokogiri'
+
+module Spree
+  module Export
+    class GoogleRss
+      prepend Spree::ServiceModule::Base
+
+      def call(settings)
+        @settings = settings
+
+        return failure(store, error: "Store with id: #{settings.spree_store_id} does not exist.") if store.nil?
+
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.rss('xmlns:g' => 'http://base.google.com/ns/1.0', 'version' => '2.0') do
+            xml.channel do
+              add_store_information_to_xml(xml)
+              store.products.active.each do |product|
+                product.variants.active.each do |variant|
+                  add_variant_information_to_xml(xml, product, variant)
+                end
+              end
+            end
+          end
+        end
+
+        success(file: builder.to_xml)
+      end
+
+      private
+
+      def store
+        return @store if defined? @store
+
+        @store ||= Spree::Store.find_by(id: @settings.spree_store_id)
+      end
+
+      def add_store_information_to_xml(xml)
+        xml.title store.name
+        xml.link store.url
+        xml.description store.meta_description
+      end
+
+      def add_variant_information_to_xml(xml, product, variant)
+        input = { product: product, variant: variant, settings: @settings, store: store }
+        result = export_required_attributes.call(input)
+
+        if result.success
+          xml.item do
+            result.value[:information]&.each do |key, value|
+              xml['g'].send(key, value)
+            end
+
+            add_optional_information(xml, product, variant)
+            add_optional_sub_attributes(xml, product, variant)
```

Comment:
I saw your comment about split services into smaller and independent, do we have any settings attribute to decide if we want to send `optional` or `sub_attributes` or those will be sent always ?

Reply:
There is no setting, but if nothing is added, than by default they will just return empty hash, and nothing will be added to xml. Do you think something like that should be added? It would definetly mean less calls to services, especially for sub_attributes.